src/main/java/nl/novi/marijana/les12huiswerkservices/services/TelevisionService.java

```diff
@@ -1,44 +1,130 @@
 import nl.novi.marijana.les12huiswerkservices.repositories.TelevisionRepository;
+import nl.novi.marijana.les12huiswerkservices.repositories.WallBracketRepository;
 import org.springframework.stereotype.Service;
 import org.springframework.web.bind.annotation.RequestBody;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
 
+//*****RELATIONS STEP 3:
+//1. add WallBracketRepository wallBracketRepository as dependency injection
+//1.a. now we can reach the repositories of both television & wallBracket
+//2. go to TelevisionController to create @PostMapping request
 @Service
 public class TelevisionService {
-
     private final TelevisionRepository televisionRepository;
+    private final WallBracketRepository wallBracketRepository;
+    private final RemoteControllerRepository remoteControllerRepository;
 
     //1. create constructor with dependency injection referring to repository class,
     // ---this way I force SpringBoot to instantiate televisionRepository object (above)
-    public TelevisionService(TelevisionRepository televisionRepository) {
+    public TelevisionService(TelevisionRepository televisionRepository, WallBracketRepository wallBracketRepository, RemoteControllerRepository remoteControllerRepository) {
         this.televisionRepository = televisionRepository;
+        this.wallBracketRepository = wallBracketRepository;
+        this.remoteControllerRepository = remoteControllerRepository;
     }
 
     //2. create methods that are "linked" to mapping methods and dtos in the controller and to the entities in the repository
-    //De Service maakt gebruik van de gegevens die we via de Controller doorkrijgen van de Dtos
+
+    //********MANY-TO-MANY RELATION: create television (post-request)
+    public TelevisionDto createTelevision(TelevisionDto tvDto) {
+        Television tv = transferToTv(tvDto);
+        televisionRepository.save(tv);
+        return transferToTvDto(tv);
+    }
+
+    //transfer dto to entity - HERE IT'S GOING WRONG(LINE 166)
+    public Television transferToTv(TelevisionDto tvDto) {
+        Television tv = new Television();
+        tv.setName(tvDto.getName());
+        tv.setAvailableSize(tvDto.getAvailableSize());
+        tv.setType(tvDto.getType());
+        tv.setBluetooth(tvDto.getBluetooth());
+        tv.setPrice(tvDto.getPrice());
+        tv.setHdr(tvDto.getHdr());
+        tv.setScreenType(tvDto.getScreenType());
+        //...the rest of the fields
+
+        //***loop to connect the right tvs to the right wall brackets:
+
+        //line 60 is where i'm getting the null pointer exception(tvDto.wallBracketIds)
+       for(Long id : tvDto.wallBracketIds) {
+            //here we are taking the wallBracket entity out of the wallBracketRepository
+           Optional<WallBracket> optionalWallBracket = wallBracketRepository.findById(id);
+           if(optionalWallBracket.isPresent()) {
+               tv.getWallBrackets().add(optionalWallBracket.get());
```

Comment:
Deze code ziet er op het oog goed uit, behalve dat je hier niet juist de waalbracket in tv zet. Ik kan je niet zeggen waarom je op regel 60 een nullpointerexception krijgt

Reply:
Dat gezegd hebbende, het is makkelijker om een "assignwallbrackettotelevision" methode te maken, net zoals je een "assignremotecontrollertotelevision" methode hebt.